tests/python/unittest/test_ndarray.py

```diff
@@ -1292,7 +1358,9 @@ def test_setitem_autograd(np_array, index):
         try:
             with mx.autograd.record():
                 x[index] = y
-                assert False  # should not reach here
+                # TODO: figure out why `None` doesn't raise
```

Comment:
I guess it's because this case is equivalent to `y.copyto(x)`, where `x` is only passed into the autograd recorder as a pure output, while in other cases, `x` is passed as both an input and an output tensor. Therefore, it's fine for skipping this case.

Reply:
Okay, thanks for the pointer. I'll change the comment. There are a couple of cases where `None` is in the indexing tuple, and those raise as expected.